File: app/src/main/java/com/uae_barq/uaebarqtasks/app_config/AppConfig.java
package com.uae_barq.uaebarqtasks.app_config;

import android.app.Application;
import android.content.ContextWrapper;
import android.content.res.Configuration;

import androidx.annotation.NonNull;

import com.pixplicity.easyprefs.library.Prefs;

public class AppConfig extends Application {

    private static AppConfig mInstance;

    public static synchronized AppConfig getInstance() {
        return mInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        // Initialize the Prefs class
        new Prefs.Builder()
                .setContext(this)
                .setMode(ContextWrapper.MODE_PRIVATE)
                .setPrefsName(getPackageName())
                .setUseDefaultSharedPreference(true)
                .build();

        mInstance = this;

    }

    @Override
    public void onTerminate() {
        super.onTerminate();
    }

    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }
}

File: app/src/main/java/com/uae_barq/uaebarqtasks/kotlin/app_config/AppConfig.kt
package com.uae_barq.uaebarqtasks.kotlin.app_config

import android.app.Application

import com.crashlytics.android.Crashlytics

import io.fabric.sdk.android.Fabric

class AppConfig : Application() {

    private var mInstance: AppConfig? = null


    /**
     * Synchronized methods enable a simple strategy for preventing thread interference and memory
     * consistency errors: if an object is visible to more
     * than one thread,
     * all reads or writes to that object's variables are done through synchronized methods.
     * (An important exception: final fields, which cannot be modified after the object is constructed,
     * can be safely read through non-synchronized methods, once the object is constructed) This strategy
     * is effective,
     * @return
     */
    @Synchronized
    fun getInstance(): AppConfig? {
        return mInstance
    }

    override fun onCreate() {
        super.onCreate()

        //Installing crash report
        Fabric.with(this, Crashlytics())

        mInstance = this
    }

}
